src/formatDate.js

```diff
@@ -7,8 +7,39 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // Розділяємо дату на частини за допомогою роздільника у форматі fromFormat
+  const separator = fromFormat.pop(); // Останній елемент - це роздільник
+  const dateParts = date.split(separator);
+
+  // Створюємо об'єкт, що містить частини дати відповідно до fromFormat
+  const dateObj = {};
+
+  fromFormat.forEach((part, i) => {
+    dateObj[part] = dateParts[i];
```

Comment:
When assigning date parts to the `dateObj`, it's important to ensure that the `dateParts` array has the same length as the `fromFormat` array. This is guaranteed by the task conditions, but if the function is to be more robust, you might consider adding a check here to avoid potential issues in case of future changes to the input validation.